python/cudf/cudf/comm/gpuarrow.py

```diff
@@ -141,91 +137,6 @@ def _make_dictionary_series(self):
         return self._series.copy(deep=False)
 
 
-def gpu_view_as(nbytes, buf, dtype, shape=None, strides=None):
-    ptr = numba.cuda.cudadrv.driver.device_pointer(buf.to_numba())
-    arr = rmm.device_array_from_ptr(ptr, nbytes // dtype.itemsize, dtype=dtype)
-    arr.gpu_data._obj = buf
-    return arr
-
-
-def make_device_arrays(array):
-    buffers = array.buffers()
-    dtypes = [np.dtype(np.int8), None, None]
-
-    if pa.types.is_list(array.type):
-        dtypes[1] = np.dtype(np.int32)
-    elif pa.types.is_string(array.type) or pa.types.is_binary(array.type):
-        dtypes[2] = np.dtype(np.int8)
-        dtypes[1] = np.dtype(np.int32)
-    elif not pa.types.is_dictionary(array.type):
-        dtypes[1] = arrow_to_pandas_dtype(array.type)
-    else:
-        dtypes[1] = arrow_to_pandas_dtype(array.type.index_type)
-
-    if buffers[0] is not None:
-        buf = CudaBuffer.from_buffer(buffers[0])
```

Comment:
This was already totally doing a Host --> Device here...

Reply:
I don't think it's doing a Host -> Device copy here. Arrow Array's `buffers()` method returns a list of opaque Arrow Buffers whose pointers can be either host or device. This `from_buffer()` call unwraps[1] the opaque pointer and returns a concrete device buffer type that we use to get the device pointer via numba a bit later.

1. https://github.com/apache/arrow/blob/apache-arrow-0.15.0/python/pyarrow/_cuda.pyx#L402